runtime/ai/developer_agent.go

```diff
@@ -138,14 +145,24 @@ You should not make many changes at a time. Carefully consider the minimum chang
 }
 
 func (t *DeveloperAgent) userPrompt(args *DeveloperAgentArgs) string {
-	if !args.InitProject {
-		return args.Prompt
+	var prompt strings.Builder
+
+	// Add context about initialization if applicable
+	if args.InitProject {
+		prompt.WriteString("The project is currently empty apart from a few boilerplate files (like rill.yaml).\n")
+		prompt.WriteString("The user has asked you to help them set up their initial project based on the following instructions: ")
+	}
+
+	// Add context about current file if available
+	if args.CurrentFilePath != "" {
+		prompt.WriteString(fmt.Sprintf("The user is currently viewing/editing the file: %s\n", args.CurrentFilePath))
+		prompt.WriteString("Their request may relate to this file.\n\n")
```

Comment:
In the analyst agent, we put contextual info in the system prompt (not the user prompt). I'm not really sure which is more correct, but we should probably be consistent. (On reflection, the existing `InitProject` handling should probably also have been added to the system prompt.)

See example here: https://github.com/rilldata/rill/blob/e03c48f9466e0a35f2b026a9dd25f8fe5f251a44/runtime/ai/analyst_agent.go#L224-L228